module-2/backend/week_two.md

```diff
@@ -5,37 +5,118 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. At a high level, what is ActiveRecord? What does it do/allow you to do?
+
+  A: ActiveRecord is an ORM that allows you to manipulate a SQL database more easily.
+  
 2. What is a migration?
+
+  A: A migration updates your schema with the changes that you want to make to your database structure.
+    
 3. How does a table relate to a model?
+
+  A: A model represents a table in a database.
+    
 4. What kind of methods are `belongs_to`, and `has_many`? (i.e. class or instance) Give an example.
+
+  A: Class method - e.g. belongs_to :tasks.
+  
 5. What do they allow you to do?
+
+  A: They connect tables in a database through a foreign key.
+  
 6. What's the difference between agile workflow and waterfall method?
+
+  A: Agile workflow breaks the work up into chunks/iterations, tests and refactors each iteration along the way. Waterfall method does each piece of work separately (design, code, test, etc.) one after another.
+  
 7. What is the difference between `#new` and `#create`?
+
+  A: "new" initializes an object but doesn't save, "create" initializes and saves.
+  
 8. At a basic level, what does cURL allow you to do?
+
+  A: It allows you to put information into a database through the command line.
+  
 9. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+
+  A: many-to-many
+  
 10. Define foreign key, primary key, and schema.
+  
+    A: Foreign Key: A foreign key in one table refers to the primary key of another, defines relationships between tables.
+       Primary Key: The key that identifies a table without referring to any of its attributes.
+       Schema:  The architecture used to build a database.
+       
 11. Describe the relationship between a foreign key on one table and a primary key on another table.
+
+   A: It defines the relationship between the tables - when a foreign key is included, it means that the table with the foreign key belongs to the table with the primary key it refers to.
+  
 12. What are the parts of an HTTP response?
+
+  A: Header, response code, body
+
 13. `Rack::Test` allows us to test our controllers in isolation. What are some of the methods it gives us to simulate the request/response cycle?
-14. Describe some techniques to make our Sinatra views more DRY. Give an example of when you would use these techniques.
 
+   A: I don't know what that is! :)
```

Comment:
Ha! Good catch. We didn't teach controller testing this time around. You'll get it with Josh in Mod 3.